ui/dashboards/src/utils/time-range-params.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useMemo, useEffect, useRef } from 'react';
+import { useQueryParams, DecodedValueMap, StringParam } from 'use-query-params';
+import { getUnixTime, sub } from 'date-fns';
+import {
+  TimeRangeValue,
+  AbsoluteTimeRange,
+  toAbsoluteTimeRange,
+  isRelativeTimeRange,
+  isDurationString,
+  RelativeTimeRange,
+  DurationString,
+  isAbsoluteTimeRange,
+  parseDurationString,
+} from '@perses-dev/core';
+import { useSelectedTimeRange } from '../context';
+
+const timeRangeQueryConfig = {
+  start: StringParam,
+  end: StringParam,
```

Comment:
Rather than `StringParam` should this be some kind of custom param type to avoid some of the conversion logic below?

Reply:
Yeah good call, this is updated now (see [TimeRangeParam](https://github.com/perses/perses/blob/1b5377230d8a9cc04796019f4c084dc3e574a0ce/ui/dashboards/src/utils/time-range-params.ts#L119)) which simplified things a ton